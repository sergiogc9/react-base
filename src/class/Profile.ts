export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
