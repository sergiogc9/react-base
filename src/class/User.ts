export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
