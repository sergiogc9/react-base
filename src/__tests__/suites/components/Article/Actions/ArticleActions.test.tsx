import { mount, shallow } from "enzyme";
import React from "react";
import ArticleActions from "@src/components/Article/Actions/ArticleActions";
import { DocumentObject } from "@src/class/Document";
import GA from '@src/lib/googleAnalytics';

let onToggleAllDocumentsMock: any;
let onRemoveDocumentBulkMock: any;
let onSetTagsCategoryMock: any;
const documentSources: DocumentObject[] | null = [
	{
		place: null,
		date_from_provider: "",
		tags: null,
		queries: [],
		category: null,
		media: null,
		image_url: "",
		image: null,
		page_number: "",
		brand_associated: "",
		category_id: "",
		country: null,
		issue_number: "",
		page_occupation: 0,
		author: {
			id: 1506856080,
			name: "Igor Giannasi"
		},
		content: "content",
		date: "2019-04-04T06:30:57Z",
		id: "15543594571385654087",
		language: {
			code: "pt",
			id: "7",
			name: "Portuguese"
		},
		link: "http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==",
		location: {
			continent: {
				id: "02",
				name: "South America"
			},
			country: {
				id: "0208",
				iso: "BR",
				name: "Brazil"
			}
		},
		provider: "News",
		rank: {
			audience: 1311327,
			similarweb_monthly_visits: 61423055,
		},
		social: {},
		source: {
			id: "1233730921",
			title: "http://www.terra.com.br",
			url: "http://www.terra.com.br",
			name: ""
		},
		title: "Peça &#39;O Aniversário de Jean Lucca&#39; retrata a &#39;cultura da indiferença&#39; na sociedade brasileira",
	}
];

describe("Search Actions", () => {
	let wrapper: any;

	beforeEach(() => {
		onToggleAllDocumentsMock = jest.fn();
		onRemoveDocumentBulkMock = jest.fn();
		onSetTagsCategoryMock = jest.fn();
		wrapper = mount(
			<ArticleActions
				documentSources={documentSources}
				documentsChecked={{ 15543594571385654087: true }}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
	});

	it("Test component snapshot without results checked", () => {
		wrapper = mount(
			<ArticleActions
				documentSources={[]}
				documentsChecked={{}}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with results checked", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test check all documents event", () => {
		expect(onToggleAllDocumentsMock).toHaveBeenCalledTimes(0);
		wrapper.find('#documentSelectAll').at(0).props().onChange();
		expect(onToggleAllDocumentsMock).toHaveBeenCalledTimes(1);
	});

	it("Test remove action selected", () => {
		expect(wrapper.state().removeAction).toBe(false);
		wrapper.find('.document-actions-remove').at(0).simulate('click');
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.state().removeAction).toBe(true);
	});

	it("Test remove action close", () => {
		wrapper.find('.document-actions-remove').at(0).simulate('click');
		expect(wrapper.state().removeAction).toBe(true);
		wrapper.find('.confirm-cancel').at(0).simulate('click');
		expect(wrapper.state().removeAction).toBe(false);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test remove document", () => {
		wrapper.find('.document-actions-remove').at(0).simulate('click');
		expect(onRemoveDocumentBulkMock).toHaveBeenCalledTimes(0);
		wrapper.find('.confirm-accept').at(0).simulate('click');
		expect(onRemoveDocumentBulkMock).toHaveBeenCalledTimes(1);
	});

	it("Test categorize action selected", () => {
		wrapper = shallow(
			<ArticleActions
				documentSources={documentSources}
				documentsChecked={{ 15543594571385654087: true }}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
		expect(wrapper.state().categorizeAction).toBe(false);
		wrapper.find('.document-actions-categorization').at(0).simulate('click');
		expect(wrapper.debug()).toMatchSnapshot();
		expect(wrapper.state().categorizeAction).toBe(true);
	});

	it("Test onAddTagsCalled and onRemoveTag and setCategory", () => {
		const set = new Set();
		wrapper = shallow(
			<ArticleActions
				documentSources={documentSources}
				documentsChecked={{ 15543594571385654087: true }}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
		wrapper.find('.document-actions-categorization').at(0).simulate('click');
		wrapper.find("#categorizeActionContent").children().at(1).props().onAddTags(['tagTest']);
		set.add('tagTest');
		expect(wrapper.state().tags).toEqual(set);
		wrapper.find("#categorizeActionContent").children().at(1).props().onRemoveTag('tagTest');
		set.delete('tagTest');
		expect(wrapper.state().tags).toEqual(set);
		wrapper.find("#categorizeActionContent").children().at(1).props().onSetCategory('00110011');
		expect(wrapper.state().category).toEqual("00110011");
	});

	it("Test onSetTagsCategory confirm action", () => {
		wrapper = shallow(
			<ArticleActions
				documentSources={documentSources}
				documentsChecked={{ 15543594571385654087: true }}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
		expect(onSetTagsCategoryMock).toHaveBeenCalledTimes(0);
		wrapper.find('.document-actions-categorization').at(0).simulate('click');
		wrapper.find('.confirm-accept').at(0).simulate('click');
		expect(onSetTagsCategoryMock).toHaveBeenCalledTimes(1);
	});

	it("Test onSetTagsCategory dismiss action", () => {
		wrapper = shallow(
			<ArticleActions
				documentSources={documentSources}
				documentsChecked={{ 15543594571385654087: true }}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
		wrapper.find('.document-actions-categorization').at(0).simulate('click');
		expect(wrapper.state().categorizeAction).toBe(true);
		wrapper.find('.confirm-cancel').at(0).simulate('click');
		expect(wrapper.state().categorizeAction).toBe(false);
	});

	it("Test update props", () => {
		wrapper.setProps({ documentsChecked: {} });
		wrapper.update();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test remove documents tracker", () => {
		GA.trackEvent = jest.fn();
		wrapper = shallow(
			<ArticleActions
				documentSources={documentSources}
				documentsChecked={{ 15543594571385654087: true }}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		wrapper.find('.document-actions-remove').at(0).simulate('click');
		wrapper.find('.confirm-accept').at(0).simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
		expect(GA.trackEvent).toHaveBeenCalledWith({ category: 'Feed results', action: 'Remove mention' });

		wrapper.setProps({ ...wrapper.props, documentsChecked: { 15543594571385654087: true, 1554359457138565890: true } });
		wrapper.find('.confirm-accept').at(0).simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(2);
		expect(GA.trackEvent).toHaveBeenCalledWith({ category: 'Feed results', action: 'Remove multiple mentions' });

	});

	it("Test onSetTagsCategory trackers", () => {
		const set = new Set();
		set.add('tagTest');
		GA.trackEvent = jest.fn();

		wrapper = shallow(
			<ArticleActions
				documentSources={documentSources}
				documentsChecked={{ 15543594571385654087: true }}
				onSetTagsCategory={onSetTagsCategoryMock}
				onToggleAllDocuments={onToggleAllDocumentsMock}
				onRemoveDocumentBulk={onRemoveDocumentBulkMock}
			/>
		);
		wrapper.find('.document-actions-categorization').at(0).simulate('click');
		wrapper.find('.confirm-accept').at(0).simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);

		wrapper.setState({ ...wrapper.state, tags: set });
		wrapper.find('.confirm-accept').at(0).simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);

		wrapper.setState({ ...wrapper.state, category: "123456" });
		wrapper.find('.confirm-accept').at(0).simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(3);
	});
});
