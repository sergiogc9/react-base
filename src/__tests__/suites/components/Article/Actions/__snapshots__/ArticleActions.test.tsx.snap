// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Actions Test categorize action selected 1`] = `
"<Collapse collapsed={false} animate={true} springConfig={{...}} minHeight={0}>
  <div id=\\"articleActions\\">
    <div id=\\"actionsButtons\\">
      <SelectionControl id=\\"documentSelectAll\\" name=\\"documentSelectAll\\" aria-label=\\"documentSelectAll\\" type=\\"checkbox\\" onChange={[Function: onChange]} checked={true} className=\\"checkbox-green document-actions-select-checkbox\\" uncheckedCheckboxIcon={{...}} checkedCheckboxIcon={{...}} checkedRadioIcon={{...}} uncheckedRadioIcon={{...}} />
      <withInk(withTooltip(Button)) raised={true} className=\\" document-actions-remove\\" onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
        results.bulk_action.remove.value
      </withInk(withTooltip(Button))>
      <withInk(withTooltip(Button)) raised={true} className=\\"action-selected document-actions-categorization\\" onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
        results.bulk_action.categorize
      </withInk(withTooltip(Button))>
      <span>
        1
         
        results.bulk_selection.articles_count
      </span>
    </div>
    <div id=\\"actionsConfirm\\">
      <Collapse collapsed={false} animate={true} springConfig={{...}} minHeight={0}>
        <div className=\\"confirm-action-content\\" id=\\"categorizeActionContent\\">
          <div className=\\"confirm-info\\">
            <span className=\\"confirm-info-action\\">
              results.bulk_action.categorize
            </span>
            <span className=\\"confirm-info-selected\\">
              1
               
              results.bulk_action.remove.confirm.articles
            </span>
          </div>
          <Connect(Categorization) tags={{...}} onAddTags={[Function: onAddTags]} onRemoveTag={[Function: onRemoveTag]} onSetCategory={[Function: onSetCategory]} />
          <div className=\\"confirm-buttons\\">
            <withInk(withTooltip(Button)) raised={true} className=\\"confirm-accept\\" onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
              results.bulk_action.remove.confirm.yes
            </withInk(withTooltip(Button))>
            <span className=\\"confirm-conjuntion\\">
              results.bulk_action.remove.confirm.or
            </span>
            <span className=\\"confirm-cancel\\" onClick={[Function: onClick]}>
              results.bulk_action.remove.confirm.cancel
            </span>
          </div>
        </div>
      </Collapse>
    </div>
  </div>
</Collapse>"
`;

exports[`Search Actions Test component snapshot with results checked 1`] = `"<div id=\\"articleActions\\" style=\\"overflow: hidden;\\"><div id=\\"actionsButtons\\"><div class=\\"md-selection-control-container checkbox-green document-actions-select-checkbox\\"><input id=\\"documentSelectAll\\" type=\\"checkbox\\" class=\\"md-selection-control-input\\" name=\\"documentSelectAll\\" aria-hidden=\\"true\\" value=\\"\\" checked=\\"\\"><label for=\\"documentSelectAll\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-label=\\"documentSelectAll\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><button type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block  document-actions-remove\\"><div class=\\"md-ink-container\\"></div>results.bulk_action.remove.value</button><button type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block  document-actions-categorization\\"><div class=\\"md-ink-container\\"></div>results.bulk_action.categorize</button><span>1 results.bulk_selection.articles_count</span></div><div id=\\"actionsConfirm\\"></div></div>"`;

exports[`Search Actions Test component snapshot without results checked 1`] = `""`;

exports[`Search Actions Test remove action close 1`] = `"<div id=\\"articleActions\\" style=\\"\\"><div id=\\"actionsButtons\\"><div class=\\"md-selection-control-container checkbox-green document-actions-select-checkbox\\"><input id=\\"documentSelectAll\\" type=\\"checkbox\\" class=\\"md-selection-control-input\\" name=\\"documentSelectAll\\" aria-hidden=\\"true\\" value=\\"\\" checked=\\"\\"><label for=\\"documentSelectAll\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-label=\\"documentSelectAll\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><button type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block  document-actions-remove\\"><div class=\\"md-ink-container\\"></div>results.bulk_action.remove.value</button><button type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block  document-actions-categorization\\"><div class=\\"md-ink-container\\"></div>results.bulk_action.categorize</button><span>1 results.bulk_selection.articles_count</span></div><div id=\\"actionsConfirm\\"></div></div>"`;

exports[`Search Actions Test remove action selected 1`] = `"<div id=\\"articleActions\\" style=\\"\\"><div id=\\"actionsButtons\\"><div class=\\"md-selection-control-container checkbox-green document-actions-select-checkbox\\"><input id=\\"documentSelectAll\\" type=\\"checkbox\\" class=\\"md-selection-control-input\\" name=\\"documentSelectAll\\" aria-hidden=\\"true\\" value=\\"\\" checked=\\"\\"><label for=\\"documentSelectAll\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-label=\\"documentSelectAll\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><button type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block action-selected document-actions-remove\\"><div class=\\"md-ink-container\\"></div>results.bulk_action.remove.value</button><button type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block  document-actions-categorization\\"><div class=\\"md-ink-container\\"></div>results.bulk_action.categorize</button><span>1 results.bulk_selection.articles_count</span></div><div id=\\"actionsConfirm\\"><div class=\\"confirm-action-content\\" id=\\"removeActionContent\\"><div class=\\"confirm-info\\"><span class=\\"confirm-info-action\\">results.bulk_action.remove.value</span><span class=\\"confirm-info-selected\\">1 results.bulk_action.remove.confirm.articles</span></div><div class=\\"confirm-message confirm-message-remove\\">results.bulk_action.remove.confirm.question</div><div class=\\"confirm-buttons\\"><button type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block confirm-accept\\"><div class=\\"md-ink-container\\"></div>results.bulk_action.remove.confirm.yes</button><span class=\\"confirm-conjuntion\\">results.bulk_action.remove.confirm.or</span><span class=\\"confirm-cancel\\">results.bulk_action.remove.confirm.cancel</span></div></div></div></div>"`;

exports[`Search Actions Test update props 1`] = `""`;
